### ISSUE TITLE
Consider introducing git commit standards

### ISSUE THREAD

**Issue by jbesraa** [stratum-mining/stratum#978]
Currently our git history looks very different between the different users. It would be great if we could enforce some standards so its easier to read it.  

I propose to use the same Contribution Flow section from rust-lightning: https://github.com/lightningdevkit/rust-lightning/blob/main/CONTRIBUTING.md

```
Contribution Workflow
---------------------

The codebase is maintained using the "contributor workflow" where everyone
without exception contributes patch proposals using "pull requests". This
facilitates social contribution, easy testing and peer review.

To contribute a patch, the workflow is as follows:

  1. Fork Repository
  2. Create topic branch
  3. Commit patches

In general commits should be atomic and diffs should be easy to read.
For this reason do not mix any formatting fixes or code moves with
actual code changes. Further, each commit, individually, should compile
and pass tests, in order to ensure git bisect and other automated tools
function properly.

When adding a new feature, like implementing a spec object, thought
must be given to the long term technical debt. Every new features should
be covered by functional tests.

When refactoring, structure your PR to make it easy to review and don't
hesitate to split it into multiple small, focused PRs.

The Minimum Supported Rust Version (MSRV) currently is 1.63.0 (enforced by
our GitHub Actions). 

Commits should cover both the issue fixed and the solution's rationale. These
[guidelines](https://chris.beams.io/posts/git-commit/) should be kept in mind.

To facilitate communication with other contributors, the project is making use
of GitHub's "assignee" field. First check that no one is assigned and then
comment suggesting that you're working on it. If someone is already assigned,
don't hesitate to ask if the assigned party or previous commenters are still
working on it if it has been awhile.

Any changes that have nontrivial backwards compatibility considerations should
have an entry added in the `pending_changelog` folder which includes the
CHANGELOG entries that should be added in the next release.
```

**Comment by pythcoiner** [comment]
seems contribution workflow already [here](https://github.com/stratum-mining/stratum/blob/main/CONTRIBUTING.md#contribution-workflow)

**Comment by jbesraa** [comment]
Yea you are right. It is though missing the git commit guidelines: 

"Commits should cover both the issue fixed and the solution's rationale. These [guidelines](https://chris.beams.io/posts/git-commit/) should be kept in mind."

**Comment by rrybarczyk** [comment]
Can we enforce a branch naming convention?

I prefer `<issue number-short-present-tense-description>`, for example `813-standardize-cli`.

**Comment by jbesraa** [comment]
Not sure if that is something worth forcing as its not really created in the `stratum` repo but in the forked repo

**Comment by pythcoiner** [comment]
> Not sure if that is something worth forcing as its not really created in the `stratum` repo but in the forked repo

agree

**Comment by Fi3** [comment]
> Can we enforce a branch naming convention? Or is that out of scope for the issue?
> 
> I prefer `<issue number-short-present-tense-description>`, for example `813-standardize-cli`.

I don't like use issue number (that are a github concept) in commits. IMO git log should not depend on third parties like github.

**Comment by rrybarczyk** [comment]
> > Can we enforce a branch naming convention? Or is that out of scope for the issue?
> > I prefer `<issue number-short-present-tense-description>`, for example `813-standardize-cli`.
> 
> I don't like use issue number (that are a github concept) in commits. IMO git log should not depend on third parties like github.

That's fair. But we do use github, and if we used a different platform, that platform would still have issue numbers.

Alternatively, we could prepend the year and month like `yyyy-mm-branch-name`. I believe this is how rust lightning does it. If we do that, I would like to really enforce the use of linking issues and PRs with github's "Development" feature:

<img width="260" alt="Screen Shot 2024-07-02 at 12 09 16 PM" src="https://github.com/stratum-mining/stratum/assets/29670338/24d32ccd-5d4e-4076-814f-4bccbffd9a60">

**Comment by Fi3** [comment]
> > > Can we enforce a branch naming convention? Or is that out of scope for the issue?
> > > I prefer `<issue number-short-present-tense-description>`, for example `813-standardize-cli`.
> > 
> > 
> > I don't like use issue number (that are a github concept) in commits. IMO git log should not depend on third parties like github.
> 
> That's fair. But we do use github, and if we used a different platform, that platform would still have issue numbers.
> 
> Alternatively, we could prepend the year and month like `yyyy-mm-branch-name`. I believe this is how rust lightning does it. If we do that, I would like to really enforce the use of linking issues and PRs with github's "Development" feature:
> <img alt="Screen Shot 2024-07-02 at 12 09 16 PM" width="260" src="https://private-user-images.githubusercontent.com/29670338/345111272-24d32ccd-5d4e-4076-814f-4bccbffd9a60.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk5OTc0NTQsIm5iZiI6MTcxOTk5NzE1NCwicGF0aCI6Ii8yOTY3MDMzOC8zNDUxMTEyNzItMjRkMzJjY2QtNWQ0ZS00MDc2LTgxNGYtNGJjY2JmZmQ5YTYwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzAzVDA4NTkxNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE1YjVhYWQwNjI4NzU3ZWFjOTA5ODI4YTc4NzUyNjZmYjA5NTAyYzFkMWQxMjAzNzFkMWI0NzdlN2NiNjE1NDEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.U8v3S_W2EoBN7MUjfPO9kAYL8-CvcCHM0Ryy_Y1PYxE">

yep my concern is not the fact that we use issue. But the fact that we refer to something that is hosted in a 3th party service that as fa as I know could close our account tomorrow.

**Comment by rrybarczyk** [comment]
@Fi3 sounds good. Which convention do you like? We can follow rust-lightning and do `yyyy-mm-branch-name`, or we can include the day like `yyyy-mm-dd-branch-name`? Or do you prefer something else?

**Comment by Fi3** [comment]
> @Fi3 sounds good. Which convention do you like? We can follow rust-lightning and do `yyyy-mm-branch-name`, or we can include the day like `yyyy-mm-dd-branch-name`? Or do you prefer something else?

no strong opinion

**Comment by pavlenex** [comment]
@jbesraa this was done as part of #978 should we close it, I saw now you're logging things here #1098.

**Comment by rrybarczyk** [comment]
> > @Fi3 sounds good. Which convention do you like? We can follow rust-lightning and do `yyyy-mm-branch-name`, or we can include the day like `yyyy-mm-dd-branch-name`? Or do you prefer something else?
> 
> no strong opinion

Let's do `yyyy-mm-branch-name`.

**Comment by jbesraa** [comment]
> @jbesraa this was done as part of #978 should we close it? I saw now you're logging things here #1098.

This issue was addressed actually here https://github.com/stratum-mining/stratum/pull/991 so we can close this and move any further discussions/new ideas to #1098

**Comment by rrybarczyk** [comment]
> > @jbesraa this was done as part of #978 should we close it? I saw now you're logging things here #1098.
> 
> This issue was addressed actually here #991 so we can close it and move any further discussions/new ideas to #1098

Got it, closing in favor of #978.
